stats/src/main/java/io/airlift/stats/JmxGcMonitor.java

```diff
@@ -65,9 +65,16 @@ public class JmxGcMonitor
 
     private final TimeStat minorGc = new TimeStat();
 
+    private final long heapMemoryLimit;
+
     @GuardedBy("this")
     private long lastGcEndTime = System.currentTimeMillis();
 
+    public JmxGcMonitor()
+    {
+        this.heapMemoryLimit = ManagementFactory.getMemoryMXBean().getHeapMemoryUsage().getMax();
```

Comment:
This can move to the initializer

Reply:
Actually, I don't think this should be a constant, as the heap size can change. This should be fetched at the time of usage.

Reply:
I am not sure what to do here. It is not uncommon for Major GCs to occur before the heap has reached it's upper bound. As best I can tell, Max can be [unspecified](https://docs.oracle.com/javase/8/docs/api/java/lang/management/MemoryUsage.html), in which case Max will return -1. So perhaps what should be done is only use this when it is defined. Thoughts?

Reply:
We should call this every time, rather than making it a constant.

Reply:
PR updated.